ChangeOneSubscriptionOwnership/handler.ts

```diff
@@ -0,0 +1,149 @@
+  );
+
+/*
+ * The function purpose is to update the subscription owner id to APIM
+ * We need the complete target id PATH: /subscriptions/XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX/resourceGroups/XXXXXXX/providers/Microsoft.ApiManagement/service/XXXXXXX/users/users/XXXXXXXXXXXXXXXXXXXXXXXXXX
+ */
+export const updateApimSubscription = (
+  config: IDecodableConfigAPIM,
+  apimClient: ApiManagementClient
+) => (
+  subscriptionId: NonEmptyString,
+  targetId: NonEmptyString
+): TE.TaskEither<IApimUserError, SubscriptionUpdateResponse> =>
+  pipe(
+    TE.tryCatch(
+      () =>
+        apimClient.subscription.update(
+          config.APIM_RESOURCE_GROUP,
+          config.APIM_SERVICE_NAME,
+          subscriptionId,
+          "*",
+          {
+            ownerId: targetId
+          }
+        ),
+      e => toApimUserError((e as Error).message)
+    ),
+    TE.chain(
+      flow(
+        TE.fromPredicate(
+          // Check if ownerId is available and it's the same of targetId
+          res => res.ownerId !== undefined && res.ownerId === targetId,
+          () => toApimUserError("Error on update")
+        )
+      )
+    )
+  );
+
+export const createHandler = (
+  config: IConfig,
+  apimClient: ApiManagementClient,
+  pool: Pool
+): Parameters<typeof withJsonInput>[0] =>
+  withJsonInput(
+    async (_context, subscriptionMessage): Promise<void> =>
+      // ToDo: Refactor mapLeft errors
+      pipe(
+        subscriptionMessage,
+        ClaimSubscriptionItem.decode,
+        TE.fromEither,
+        TE.mapLeft(() => E.toError("Error on decode")),
+        // Update subscription on APIM
+        TE.chain(subscriptionToMigrate =>
+          pipe(
+            updateApimSubscription(config, apimClient)(
+              subscriptionToMigrate.subscriptionId,
+              subscriptionToMigrate.targetId
+            ),
+            // TE.mapLeft(E.toError),
+            TE.orElseW(() =>
+              pipe(
+                updateSubscriptionStatusToDatabase(config, pool)(
+                  subscriptionToMigrate.subscriptionId,
+                  SubscriptionStatus.FAILED
+                ),
+                TE.mapLeft(() => E.toError(`Error on DB after APIM Fail`)),
+                TE.chainW(() => TE.left(E.toError("Error update on APIM")))
+              )
+            ),
+
+            TE.map(() => subscriptionToMigrate) // Return subscriptionToMigrate
+          )
+        ),
+        // Update subscription status on Database
+        TE.chain(subscriptionToMigrate =>
+          pipe(
+            updateSubscriptionStatusToDatabase(config, pool)(
+              subscriptionToMigrate.subscriptionId,
+              SubscriptionStatus.COMPLETED
+            ),
+            TE.mapLeft(E.toError)
+          )
+        ),
+        // We don't want to return anything
+        TE.map(_ => void 0),
+        // If we get an error we want the handler fail
+        TE.getOrElse(err => {
+          throw err;
```

Comment:
Which kind of errors do you expect here? Other said: what is that can make the handler to fail an to retry?